### ISSUE TITLE
Feature request: Improve sorting

### ISSUE THREAD

**Issue by zyp** [thomasloven/lovelace-auto-entities#47]
Hi, this is a double feature request, as there's two improvements to sorting I would like to see:

Firstly I would like to be able to sort by area.

Secondly I would like to be able to provide a list of sorting methods, so that entities that compares equally with the first method are sorted by the second method, and so on.

Combined, this will let me have entities appear room by room with each room sorted alphabetically.

**Comment by thomasloven** [comment]
In all honesty - do you remodel your house often enough to justify a dynamically updating list of rooms?

**Comment by zyp** [comment]
I'm not sure what point you're trying to make. I haven't asked for a list of rooms, I've asked for a list of entities sorted by which room they are assigned to.

**Comment by thomasloven** [comment]
What I mean is that the easiest way to do this would be to specify one area at a time:

```yaml
filter:
  include:
    - area: bedroom
      sort:
        method: name
    - area: kitchen
      sort:
        method: name
    - area: livingroom
      sort:
        method: name
```
and that you wouldn't have to change that list very often.

**Comment by jschwalbe** [comment]
I had this typed up as a new issue then github suggested it's actually already been created, so here's what I originally typed:
---
Subject: **[Request] Secondary Sorting**

@thomasloven this is soooooo amazing. you're probably sick of hearing that. (if not sick yet: sooooo super amazing! thank you!) great work!

I know very little about programming, so when I looked at the code in the repo pertaining to sorting, I was dumbfounded... would it be trivial to add **secondary sort** as an option?

Here's why: I have my sorting of window sensors based on open/closed status (open on top!) to more quickly draw the eye to what's open. That part works great, but the rest of the sensors are sorted in a pattern I haven't yet discerned (randomly??). 

![image](https://user-images.githubusercontent.com/8005921/76167884-6e9ff780-6140-11ea-8213-73038e28becc.png)

**Comment by ildar170975** [comment]
IMHO the issue can be closed since both requirements are implemented, am I right?

**Comment by thomasloven** [comment]
Not quite. Even if the original request to be able to sort by area and something else doesn't make sense for most people, there are more legitimate use cases, so I'm keeping this open in case inspiration strikes and I come up with a good way to solve this reliably.

**Comment by poldim** [comment]
I'm not a chronic rodeler but always find my list of areas that are automated forever expanding. So if I implemented something like what you show with filters above, I'd have to update this whenever I make a new area, and then replicate that to all of my dozen or so auto-entities cards.   Right now I don't filter so all of those devices change the sort order and this feature would help put some structure around that. 

A question from me is could you implent a nested sort that would go by area first and then allow you choose a sub sort (alphabetic, last_changed, etc)?

**Comment by jbienz** [comment]
I see the ability to group by area was marked 👍7 times, but it appears this issue may have been closed "completed" without the feature being implemented. Am I understanding that correctly?

While I understand that the number of areas will eventually settle not likely change over time, as a new user to HA I have been updating these regularly. The method discussed here:

https://github.com/thomasloven/lovelace-auto-entities/issues/47#issuecomment-572115934

Of adding areas manually would not only require extra effort, it would also be problematic for me right now while I am learning HA and setting up my home.

My primary desire for auto-entities was to create a dashboard section with all batteries throughout the home, grouped by area. I wanted this dashboard section to automatically update as I added new devices and new areas. I also didn't want to worry about breaking the dashboard if I renamed an area or deleted it.

Is there any possibility of reopening this feature request?
